### ISSUE TITLE
[Discussion] module type automatically generated for each module

### ISSUE THREAD

**Issue by wende** [wende/prelchemy#4]
So that:
```elm
module M exposing (..)

a : Int
b : Int -> Int
```

and when getting `M.module` (or any other suitable name) you'd get 

```elm
M.module : 
  { a : Int
  , b : Int -> Int 
  }
```

**Comment by baransu** [comment]
Can you describe use case of it?

**Comment by wende** [comment]
Generally Erlang and Elixir has modules as a first class values. You can pass them around and call their functions dynamically.
There is a developed pattern for that which is called a `behaviour` (https://elixir-lang.org/getting-started/typespecs-and-behaviours.html#behaviours)
A most common example of a behaviour is a GenServer, which is a module that MUST implement `start_link`, `init`, `handle_call`, `handle_cast`, `handle_info` and `stop` functions.
Thanks to that you can call modules passed as variables and be sure that they implement a function you are calling.
It's a neat way to implement polymorphism around modules so that you can reason of processes as an instance tightly coupled with a module, rather than having to specify its components explicitly.

Basically that's a solution to do the same, but type-safely
